sympy/physics/control/lti.py

```diff
@@ -1079,85 +1314,86 @@ def doit(self, **kwargs):
         TransferFunction((2 - s**3)*(-p + s) + (-a*p**2 - b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s)
 
         """
-        res = None
-        for arg in self.args:
-            arg = arg.doit()
-            if res is None:
-                res = arg
-            else:
-                num_ = res.num * arg.den + res.den * arg.num
-                den_ = res.den * arg.den
-                res = TransferFunction(num_, den_, self.var)
-        return res
+
+        def _SISO_doit():
+            res = None
+            for arg in self.args:
+                arg = arg.doit()
+                if res is None:
+                    res = arg
+                else:
+                    num_ = res.num * arg.den + res.den * arg.num
+                    den_ = res.den * arg.den
+                    res = TransferFunction(num_, den_, self.var)
+            return res
+
+        def _MIMO_doit():
+            _arg = (arg.doit()._expr_mat for arg in self.args)
+            res = MatAdd(*_arg, evaluate=True)
+            return TransferFunctionMatrix.from_Matrix(res, self.var)
+
+        if self.is_SISO:
+            return _SISO_doit()
+        else:
+            return _MIMO_doit()
```

Comment:
More like,
```python
        def _SISO_doit():
            _arg = (arg.doit().to_expr() for arg in self.args)
            res = Add(*_arg, evaluate=False)
            return TransferFunction.from_rational_expression(res, self.var)

        def _MIMO_doit():
            _arg = (arg.doit()._expr_mat for arg in self.args)
            res = MatAdd(*_arg, evaluate=False)
            return TransferFunctionMatrix.from_Matrix(res, self.var)
        
        return _SISO_doit() if self.is_SISO else _MIMO_doit()
```
Similarly, pass `evaluate=False` in `Series` class.

Reply:
@namannimmo10 
I don't get why `evaluate` should be `False`. If we pass `evaluate=False` there is no purpose of having `doit` as it will return the same output as input.

```
>>> pprint(Series(tfm_c, tfm_b, tfm_a), use_unicode=False)
[5*s]                 [1  s]
[---]    [5   1  ]    [-  -]
[ 1 ]    [-  ----]    [1  1]
[   ]   *[1     2]   *[    ]
[ 5 ]    [   6*s ]{t} [5  1]
[ - ]                 [-  -]
[ 1 ]{t}              [s  1]{t}

File "d:\sympy\sympy\physics\control\lti.py", line 812, in sympy.physics.control.lti.Series
Failed example:
    pprint(Series(tfm_c, tfm_b, tfm_a).doit(), use_unicode=False)  # (2 Inputs -A-> 2 Outputs) -> (2 Inputs -B-> 1 Output) -> (1 Input -C-> 2 Outputs) is equivalent to (2 Inputs -Series Equivalent-> 2 Outputs).
Expected:
    [   /   3    \    /    3    \]
    [25*\6*s  + 1/  5*\30*s  + 1/]
    [-------------  -------------]
    [        2           6*s     ]
    [     6*s                    ]
    [                            ]
    [   /   3    \    /    3    \]
    [25*\6*s  + 1/  5*\30*s  + 1/]
    [-------------  -------------]
    [        3              2    ]
    [     6*s            6*s     ]{t}
Got:
    [5*s]                 [1  s]
    [---]    [5   1  ]    [-  -]
    [ 1 ]    [-  ----]    [1  1]
    [   ]   *[1     2]   *[    ]
    [ 5 ]    [   6*s ]{t} [5  1]
    [ - ]                 [-  -]
    [ 1 ]{t}              [s  1]{t}
```

Reply:
You're right!